### ISSUE TITLE
NVIDIA Device Plugin Only Exposes One GPU Out of Two GPUs Installed on Single Node

### ISSUE THREAD

**Issue by amir-bialek** [NVIDIA/k8s-device-plugin#1020]
Hey all,

"I have an on-premises Kubernetes cluster with multiple nodes. One of these nodes is equipped with two different GPU models:
NVIDIA GeForce RTX 3090 and NVIDIA GeForce RTX 4090

When I SSH into this node and run nvidia-smi, both GPUs are properly detected and displayed. 
I have installed the NVIDIA Device Plugin using gpu-operator Helm chart (https://github.com/NVIDIA/gpu-operator/tree/main/deployments/gpu-operator).
However, only the RTX 4090 is being exposed as a resource to Kubernetes.
Here is my current configuration:
```
devicePlugin:
  config:
    name: time-slicing-config-all
    create: true
    default: "any"
    data:
      any: |-
        version: v1
        flags:
          migStrategy: none
        sharing:
          timeSlicing:
            resources:
            - name: nvidia.com/gpu
              replicas: 5
```
I have tried different type of the configuration, but it always show only one type.
Any help ?

**Comment by xingjianll** [comment]
Hi, how did you fix this?

**Comment by amir-bialek** [comment]
Hey @xingjianll 
Had to build a custom image of the nvidia-device-plugin.
See some conversation about it here:
https://github.com/NVIDIA/k8s-device-plugin/issues/424

I edited the source code, remove on the `isspec.DisableResourceNamingInConfig(logger.ToKlog, config)` function On gpu-feature-discovery, mps-control and nvidia-device-plugin. build + psuh to dockerhub,  and deployed with the original helm chart but edited the image to mine.

I did not however tested it as at the end we just bought another machine.
But when I ran it locally the line of 
```
..Customizing the 'resources' field is not yet supported in the config. Ignoring..... 
```
Did not appear in the logs, so I will assume it is working.

If you in need for this I can share more details and the full steps I did.
